uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java

```diff
@@ -610,15 +614,24 @@ private String getHostname(String instanceId) throws ControllerException {
     return hostname;
   }
 
+  private Pair<String, Integer> getHostInfo(String instanceId) throws ControllerException {
+    Map<String, Pair<String, Integer>> instanceIdAndNameMap = HelixUtils.getInstanceToHostInfoMap(_helixManager);
```

Comment:
Looks like each `getHostInfo(instanceId)` call `HelixUtils.getInstanceToHostInfoMap(_helixManager);` which queries zookeeper directly.  This could increase request latency.

Reply:
yes, it's better to store in the map to cache it